lib/oprt.gi

```diff
@@ -607,21 +607,47 @@ local   xset,surj,G,  D,  act,  fam,  filter,  hom,  i,blockacttest;
     fi;
 
     blockacttest:=function()
-    local a,b,g,p;
-      D:=List(D,Immutable); # to store sorted
-      if not ForAll( D, IsList and IsSSortedList ) then
-        return false;
-      fi;
-      for b in D do
-	for g in GeneratorsOfGroup(G) do
-	  a:=b[1]^g;
-	  p:=PositionProperty(D,x->a in x);
-	  if OnSets(b,g)<>D[p] then
-	    return false;
-	  fi;
-	od;
-      od;
-      return true;
+        #
+        # Test is D is a block system for G
+        #
+        local  x, l1, i, b, y, a, p, g;
+        D:=List(D,Immutable); 
+        if Length(D) = 0 then
+            return false;
+        fi;
+        #
+        # x will map from points to blocks
+        #
+        x:= [];
+        l1 := Length(D[1]);     
+        if l1 = 0 then
+            return false;
+        fi;        
+        for i in [1..Length(D)] do
+            b := D[i];
+            if not IsSortedList(b) or  Length(b) <> l1 then
```

Comment:
Shouldn't this use `IsSSortedList`, as in the old code?

Also, remove extra space after `or`?